### PR TITLE
Update CONTRIBUTING.md to clarify procedures and fix typos

### LINKED ISSUE #246: Update CONTRIBUTING.md to clarify procedures and fix typos
I plan to update CONTRIBUTING.md to align with updated code review guidelines (in progress) and to address a few things that I find myself repeating when responding to pull requests from first time developers. I will also fix a few typos along the way.

### PR DESCRIPTION
**Pull request by kevinrushforth**
fixes #246

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -21,39 +21,64 @@ Open an issue on our [issues list](https://bugs.openjdk.java.net) on JBS which d
...
 1. Test your changes
 
-    Run the test suite to make sure that nothing is broken. 
+    Run the test suite to make sure that nothing is broken.
 
-2. Sign the Contributor License Agreement 
+2. Rebase your changes
 
-    Ultimately, the goal is to send accepted Pull Requests upstream to the OpenJFX repository hosted at Oracle infrastructure. Therefore, please make sure you have signed the [Oracle Contributor Agreement](http://www.oracle.com/technetwork/community/oca-486395.html) . We are not asking you to give up your copyright, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the OCA once.
+    Update your local repository with the most recent code from the main [repository]((https://github.com/javafxports/openjdk-jfx)), and rebase your branch on top of the latest `develop` branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review.
```
Comment (johanvos) [inline .github/CONTRIBUTING.md]: we can probably add a link to a general github documentation page that explains the rebase command?

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -21,39 +21,64 @@ Open an issue on our [issues list](https://bugs.openjdk.java.net) on JBS which d
...
 
-3. Rebase your changes
+3. File a bug in JBS
 
-    Update your local repository with the most recent code from the main [repository]((https://github.com/javafxports/openjdk-jfx)), and rebase your branch on top of the latest `develop` branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits.  This makes them easier to review.  As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+    If there isn't already a bug filed in [JBS](https://bugs.openjdk.java.net), then please file one at [bugreport.java.com](http://bugreport.java.com/). A JBS bug ID is needed even if you have already filed an issue in the GitHub issue tracker. A GitHub issue can be used as a convenience, but JBS is the official bug database for the OpenJFX Project.
```
Comment (johanvos) [inline .github/CONTRIBUTING.md]: can everyone add the "github-bug" label?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -65,34 +90,47 @@ reviewers. If you need to pick up changes from `develop`, you can merge
...
 
-Minimum JDK 9 is required to build OpenJFX. You must have a JDK 9 installation
+JDK 11 (at a minimum) is required to build OpenJFX. You must have a JDK 11 installation
 with the environment variable `JAVA_HOME` referencing the path to Java home for
-your JDK 9 installation. By default, tests use the same runtime as `JAVA_HOME`.
-Currently OpenJFX builds are running on JDK 9 and JDK 10.
+your JDK 11 installation. By default, tests use the same runtime as `JAVA_HOME`.
+Currently OpenJFX builds are running on JDK 11 (recommended) and JDK 12.
+
+We support development in any major Java IDE (Eclipse, IntelliJ, NetBeans). IDEs can automatically configure projects based on Gradle setup.
```
Comment (johanvos) [inline .github/CONTRIBUTING.md]: I'm not sure we really "support" it -- it's possible, but we don't have that very well explained, so "support" might be replaced with "It is possible to develop in..."

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kevinrushforth) [reply .github/CONTRIBUTING.md]: Good idea.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (kevinrushforth) [reply .github/CONTRIBUTING.md]: Anyone with JBS access can do so. I added a note.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (kevinrushforth) [reply .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -21,39 +21,64 @@ Open an issue on our [issues list](https://bugs.openjdk.java.net) on JBS which d
...
 
-3. Rebase your changes
+3. File a bug in JBS
 
-    Update your local repository with the most recent code from the main [repository]((https://github.com/javafxports/openjdk-jfx)), and rebase your branch on top of the latest `develop` branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits.  This makes them easier to review.  As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+    If there isn't already a bug filed in [JBS](https://bugs.openjdk.java.net), then please file one at [bugreport.java.com](http://bugreport.java.com/). A JBS bug ID is needed even if you have already filed an issue in the GitHub issue tracker. A GitHub issue can be used as a convenience, but JBS is the official bug database for the OpenJFX Project. Once you submit a pull request, you can add the "github-bug" label on the JBS bug, along with an Issue link to the PR. If you don't have direct JBS access, one of the Project Committers will do this for you.
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply .github/CONTRIBUTING.md]: Right, I'll fix this to say "Committers or Authors".

**Comment by kevinrushforth** [comment]
For some reason, my last commit didn't trigger a CI build. It might be related to the reported [GitHub outage](https://status.github.com/messages). In any case, all this most recent commit does is add 2 words to a sentence in CONTRIBUTING.md, so we can live without the CI build.

**Comment by kevinrushforth** [comment]
> For some reason, my last commit didn't trigger a CI build. It might be related to the reported [GitHub outage](https://blog.github.com/2018-10-21-october21-incident-report/).

The outage is reported to be fixed, and the CI builds are in progress, although the Appveyor build hasn't run yet.. I'll merge this in anyway, since any failure would be due to a CI problem not a real problem.

**Pull request merged**